Movie api

egutten opened the pull request:
User stories solved for in this branch (in the context of creating a new movie entry):

1. As a user, I want to see a dropdown list of movies based on what I type in the title box, so that I do not have to remember the whole title myself.

2. As a user, I want to select the title from the dropdown and have my selection populate the title input, so that I do not have the type the title myself.

3. As an engineer, I do not want the user to be able to make duplicate movies with the same title, so that my website has variability.

4. As a user, if the movie title I selected is already taken I want to be notified of the issue, so I can pick another title.

egutten committed:
api get request working on front-end, no ui yet.

egutten committed:
visual dropdown functionality achieved; no select method yet

egutten committed:
fully functional auto-complete dropdown; no check for preexisting titles

egutten committed:
fully functional autocomplete with error handling for duplicate titles

mgutten commented on routes/movies.js:
```diff
@@ -45,10 +47,13 @@ router.post("/", middleware.isLoggedIn, function(req, res){
-			req.flash("error", "Something went wrong...");
-			res.render("movies/new");
-		} else {
+			if(err.code == 11000){
```
Did you intend to leave the node api request in the code base?

egutten replied:
I did not intend to leave the backend api packages in, and I deleted them this morning in the latest commit.

mgutten commented on public/javascript/avengers.js:
```diff
@@ -17,5 +17,39 @@ $(function() {
+	});
+});
 
+$("#results").on("click", "li.select", function(){
```
Nice!

mgutten commented on public/javascript/avengers.js:
```diff
@@ -17,5 +17,39 @@ $(function() {
+			 $("#results").show();
+			 if(value==$(that).val() && msg["Error"] == undefined) {
+				 msg["Search"].forEach (function(e){
+					 var option = document.createElement("li");
```
Not critical - should we move the HTML creation to its own function?

egutten replied:
Done last night, with the help of a genius meepies.

mgutten commented on public/javascript/avengers.js:
```diff
@@ -17,5 +17,39 @@ $(function() {
+					 option.innerHTML = e["Title"];
+					 option.classList.add("select");
+					 $("#results").append(option);
+					 $("#results").trigger("chosen:updated");
```
What’s the purpose of the trigger function? 🤷🏼‍♂️

egutten replied:
The trigger function allows the api query to refresh every time the user types different content into the title input.

mgutten reviewed (commented):
Great job!! I’m on the edge of my seat to see how it works!!

egutten committed:
deleted uncessary javascript tag, backend node references, and created renderResults function

egutten committed:
uninstalled unnecessary backend api packages

egutten commented:
I just updated the pull request. Should be good to go, but let me know!

On Mon, Jul 22, 2019 at 7:33 PM Marshall Guttenberg <
notifications@github.com> wrote:

> *@mgutten* commented on this pull request.
>
> Great job!! I’m on the edge of my seat to see how it works!!
> ------------------------------
>
> In routes/movies.js
> <https://github.com/egutten/avengers-app/pull/1#discussion_r306106139>:
>
> > @@ -45,10 +47,13 @@ router.post("/", middleware.isLoggedIn, function(req, res){
>
>  	req.body.movie.image = req.sanitize(req.body.movie.image);
>
>  	Movie.create(req.body.movie, function(err, newMovie){
>
>  		if(err){
>
> -			console.log(err);
>
> -			req.flash("error", "Something went wrong...");
>
> -			res.render("movies/new");
>
> -		} else {
>
> +			if(err.code == 11000){
>
>
> Did you intend to leave the node api request in the code base?
> ------------------------------
>
> In public/javascript/avengers.js
> <https://github.com/egutten/avengers-app/pull/1#discussion_r306106214>:
>
> >
>
> +$("#results").on("click", "li.select", function(){
>
>
> Nice!
> ------------------------------
>
> In public/javascript/avengers.js
> <https://github.com/egutten/avengers-app/pull/1#discussion_r306106465>:
>
> > +$(document).ready(function(){
>
> +	$("#title").keyup(function(){
>
> +		$("#results").empty();
>
> +		var that = this;
>
> +		var value = $(this).val();
>
> +        $.ajax({
>
> +          url: 'https://movie-database-imdb-alternative.p.rapidapi.com/?page=1&r=json&s=' + value,
>
> +          type: 'GET',
>
> +          dataType: 'json',
>
> +          beforeSend: setHeader,
>
> +		  error: function() { alert('boo!'); },
>
> +          success: function(msg) {
>
> +			 $("#results").show();
>
> +			 if(value==$(that).val() && msg["Error"] == undefined) {
>
> +				 msg["Search"].forEach (function(e){
>
> +					 var option = document.createElement("li");
>
>
> Not critical - should we move the HTML creation to its own function?
> ------------------------------
>
> In public/javascript/avengers.js
> <https://github.com/egutten/avengers-app/pull/1#discussion_r306106653>:
>
> > +		var value = $(this).val();
>
> +        $.ajax({
>
> +          url: 'https://movie-database-imdb-alternative.p.rapidapi.com/?page=1&r=json&s=' + value,
>
> +          type: 'GET',
>
> +          dataType: 'json',
>
> +          beforeSend: setHeader,
>
> +		  error: function() { alert('boo!'); },
>
> +          success: function(msg) {
>
> +			 $("#results").show();
>
> +			 if(value==$(that).val() && msg["Error"] == undefined) {
>
> +				 msg["Search"].forEach (function(e){
>
> +					 var option = document.createElement("li");
>
> +					 option.innerHTML = e["Title"];
>
> +					 option.classList.add("select");
>
> +					 $("#results").append(option);
>
> +					 $("#results").trigger("chosen:updated");
>
>
> What’s the purpose of the trigger function? 🤷🏼‍♂️
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/egutten/avengers-app/pull/1?email_source=notifications&email_token=AMAWEVHQHOIZIYHC2VT6YPDQAZUZFA5CNFSM4IF563PKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB7G4R5I#pullrequestreview-265144565>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMAWEVBX66ZLAG4SHJUYPSLQAZUZFANCNFSM4IF563PA>
> .
>

mgutten commented:
Very clean, LGTM (looks good to me)!
